mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/reports/NavigationManager.kt

```diff
@@ -0,0 +1,73 @@
+package rs.readahead.washington.mobile.views.fragment.reports
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDirections
+import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.views.fragment.reports.di.NavControllerProvider
+import javax.inject.Inject
+
+
+class NavigationManager(
+    private val navControllerProvider: NavControllerProvider,
+    val bundle: Bundle
+) {
+
+    fun navigateTo(destinationId: Int) {
+        navControllerProvider.navController.navigate(destinationId)
```

Comment:
Let's make this functions private if we are only using them inside this class?

Reply:
What do you think if  we create extension functions for NavController to encapsulate navigation logic?

Reply:
```
  fun NavController.navigateSafe(directions: NavDirections) {
        navigate(directions)
    }

    fun NavController.navigateSafe(destinationId: Int, bundle: Bundle? = null) {
        navigate(destinationId, bundle)
    }
```

Reply:
later on you can use it directly like this ->  `navControllerProvider?.navController?.navigateSafe(destinationId, bundle)`